src/App.jsx

```diff
@@ -1,24 +1,157 @@
-import './App.css';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 import HomePage from './pages/HomePage';
 import MedsPage from './pages/MedsPage';
 import MedsLogPage from './pages/MedsLogPage';
 import Footer from './components/Footer';
+import NotFound from './components/NotFound'
+import MedicationForm from './components/MedicationForm';
 
 function App() {
-  const [currentView, setCurrentView] = useState('Home');
+  // Initialize medications from local storage or set to an empty array if not found
+  const [medications, setMedications] = useState(() => {
+    const savedMedications = localStorage.getItem('medications');
+    return savedMedications ? JSON.parse(savedMedications) : [];
+  });
+
+  // Effect to load medications from local storage on mount
+  useEffect(() => {
+    const savedMedications = localStorage.getItem('medications');
+    if (savedMedications) {
+      setMedications(JSON.parse(savedMedications));
+    }
+  }, []);
+
+  // Save medications to local storage whenever they change
+  useEffect(() => {
+    localStorage.setItem('medications', JSON.stringify(medications));
+  }, [medications]);
+
+  const saveMedication = (newMedication) => {
+    setMedications(prevMedications => [...prevMedications, newMedication]);
+  };
+
+  const removeMedication = (idToRemove) => {
+    setMedications(prevMedications => prevMedications.filter(med => med.id !== idToRemove));
+  };
+
+  const medicationNames = [
+    'Humalog(Insulin Lispro)', 'Clobazam', 'Quetiapine', 'Ibuprofen', 'Lisinopril', 'Amphetamine', 'Amlodipine', 'Prednisone', 'Gabapentin', 'Cyclobenzaprine', 'Amoxicillin', 'Levothyroxine', 'Albuterol HFA', 'Benzonatate', 'Alprazolam'
+  ];
```

Comment:
In the future, it may be better for you to put arrays of data like this into their own files so that they are easy to locate and access, since you have a lot going on in your App.jsx as it is